File: 125-ValidPalindrome/125-ValidPalindrome.java
// Last updated: 9/4/2025, 6:14:45 AM
class Solution {
    public boolean isPalindrome(String s) {
        int first =0;
        int last= s.length()-1;
        boolean flag= true;
        while (first< last){
            char a= s.charAt(first);
            char b= s.charAt(last);
           if (!Character.isLetterOrDigit(a)) {
                first++;
                continue;
            }
            if (!Character.isLetterOrDigit(b)) {
                last--;
                continue;
            }

            if (Character.toLowerCase(a) != Character.toLowerCase(b)) {
                return false;
            }

            first++;
            last--;

        }
        return true;
    }
}

File: 125-ValidPalindrome/125-ValidPalindrome.py
# Last updated: 9/4/2025, 6:14:49 AM
class Solution:
    def isPalindrome(self, s: str) -> bool:
        k=""
        for i in range(len(s)):
            if s[i].isalnum():
                k=k+s[i].lower()
        if k==k[::-1]:
            return True
        else:
            return False